one_fm/utils.py

```diff
@@ -782,96 +826,160 @@ def create_leave_allocation(employee, policy_detail, leave_type_details, from_da
+        args:
+            allocation: Object of Leave Allocation
+            is_carry_forward: Boolean
+    '''
+    # Delete old ledger entry
+    frappe.db.sql("""DELETE FROM `tabLeave Ledger Entry`
+        WHERE `transaction_name`=%s""", (allocation.name))
+
+    # Create new ledger entry
+    args = {
+        'leaves': allocation.total_leaves_allocated,
+        'from_date': allocation.from_date,
+        'to_date': allocation.to_date,
+        'is_carry_forward': is_carry_forward
+    }
+    create_leave_ledger_entry(allocation, args, True)
+
+def get_new_leave_allocated_for_annual_paid_leave(allocation, leave_type):
+    '''
+        Function is used to get new leave allocated for annual paid leave
+        args:
+            allocation: Object of Leave Allocation
+            leave_type: Object of Leave Type
+        return: Fraction of Allocation
+    '''
+    new_leaves_allocated = 0
+
+    # Fetch employee annual leave allocation from employee
+    employee_annual_leave = frappe.db.get_value('Employee', allocation.employee, 'annual_leave_balance')
+    if employee_annual_leave and employee_annual_leave <= 0:
+        # Fetch employee annual leave allocation from company defaults
+        employee_annual_leave = frappe.db.get_value('Company', {"name": frappe.defaults.get_user_default("company")}, 'default_annual_leave_balance')
+
+    # calculate daily allocation factor
+    if employee_annual_leave > 0:
+        new_leaves_allocated = employee_annual_leave/365
+
+    # Set Daily Allocation from annual leave dependent leave type and reduction level
+    if leave_type.one_fm_annual_leave_allocation_reduction:
+        from erpnext.hr.doctype.leave_application.leave_application import get_leaves_for_period
+        leave_days = 0
+        if leave_type.one_fm_paid_sick_leave_type_dependent:
+            leave_days += get_leaves_for_period(allocation.employee, leave_type.one_fm_paid_sick_leave_type_dependent, allocation.from_date, allocation.to_date)
+        else:
+            leave_type_list = frappe.get_all('Leave Type', filters= {'one_fm_is_paid_sick_leave': 1}, fields= ["name"])
+            for lt in leave_type_list:
+                leave_days += get_leaves_for_period(allocation.employee, lt.name, allocation.from_date, allocation.to_date)
+
+        if leave_days > 0:
+            percent_of_reduction = 0
+            allocation_reductions = sorted(leave_type.one_fm_annual_leave_allocation_reduction, key=lambda x: x.number_of_paid_sick_leave)
+            for allocation_reduction in allocation_reductions:
+                if allocation_reduction.number_of_paid_sick_leave <= leave_days:
+                    percent_of_reduction = allocation_reduction.percentage_of_allocation_reduction
+            if percent_of_reduction > 0:
+                new_leaves_allocated -= new_leaves_allocated * (percent_of_reduction/100)
+
+    return new_leaves_allocated
+
+def get_paid_annual_leave_allocation_list(date=nowdate()):
```

Comment:
@pjamsheer , you are passing `date` as an argument to the `get_paid_annual_leave_allocation_list` method, but you aren't passing anything in the method call!

Reply:
since the definition having the argument value initialised

Reply:
👍